aptos-labs/aptos-indexer-processors
# [Draft] Create nft_orderbooks parser and processor

## Description 
* Created a new processor and parser for nft_orderbooks.
  * Currently only parsing Topaz events
  * Next I'll work on parsing other Marketplaces and parsing NFT listings and bids from write set changes
* Created BigQuery stream manager to batch and write data to BigQuery
* In order to write to BQ, we need to create a BQ table schema and protobuf schema for streaming writes. There's a way to auto-generate the BQ table schema from protobuf schema, which I'll look into later. 
* Updated config.yaml to support multiple processors and configuration states
* Created class for Token helper functions

### Data Model
https://docs.google.com/spreadsheets/d/15GdG0u-Bl-SfdRlBvYimbgjXn76kbVm1IbBUYuHCIBg/edit#gid=0

## Testing
1. Run `poetry run python run_processors.py -c config.yaml`
2. See that NFT marketplace events are being streamed to BigQuery
![Screenshot 2023-05-11 at 1 38 45 PM](https://github.com/aptos-labs/aptos-indexer-processors/assets/8248583/ec074436-767b-419f-b768-a612d93ced8b)
